modules/database/sqlite/activate.php

```diff
@@ -0,0 +1,164 @@
+		// Init the filesystem to allow copying the file.
+		global $wp_filesystem;
+		if ( ! $wp_filesystem ) {
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+		}
+
+		// Copy the file, replacing contents as needed.
+		if ( $wp_filesystem->touch( $destination ) ) {
+
+			// Get the db.copy file contents, replace placeholders and write it to the destination.
+			$file_contents = str_replace(
+				'{SQLITE_IMPLEMENTATION_FOLDER_PATH}',
+				__DIR__,
+				file_get_contents( __DIR__ . '/db.copy' )
+			);
+
+			$wp_filesystem->put_contents( $destination, $file_contents );
+		}
+	}
+
+	// As an extra safety check, bail if the current user cannot update
+	// (or install) WordPress core.
+	if ( ! current_user_can( 'update_core' ) ) {
+		return;
+	}
+
+	// Otherwise install WordPress in the SQLite database before the redirect,
+	// so that the user does not end up on the WordPress installation screen
+	// but rather remains in the Performance Lab modules screen.
+	add_filter(
+		'wp_redirect',
+		function( $redirect_location ) {
+			// If the SQLite DB already exists, simply ensure the module is
+			// active there.
+			require_once __DIR__ . '/constants.php';
+			if ( file_exists( FQDB ) ) {
+				require_once __DIR__ . '/wp-includes/sqlite/db.php';
+				wp_set_wpdb_vars();
+				global $wpdb;
+				$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", PERFLAB_MODULES_SETTING ) );
+				if ( is_object( $row ) ) {
+					$value = maybe_unserialize( $row->option_value );
+					if ( is_array( $value ) && ( ! isset( $value['database/sqlite'] ) || ! $value['database/sqlite']['enabled'] ) ) {
+						$value['database/sqlite'] = array( 'enabled' => true );
+						$wpdb->update( $wpdb->options, array( 'option_value' => maybe_serialize( $value ) ), array( 'option_name' => PERFLAB_MODULES_SETTING ) );
+					}
+				}
+				return $redirect_location;
+			}
+
+			// Get current basic setup data to install WordPress in the new DB.
+			$blog_title = get_bloginfo( 'name' );
+			$is_public  = (bool) get_option( 'blog_public' );
+			$language   = get_option( 'WPLANG' );
+			$user_email = get_option( 'admin_email' );
+			$admin_user = get_user_by( 'email', $user_email );
+			if ( ! $admin_user ) {
+				$admin_user = wp_get_current_user();
+			}
+
+			// If the current user is not the admin email user, look up the
+			// data for the current user. Additionally, attempt to keep them
+			// logged-in by retaining their current session. Depending on the
+			// site configuration, this is not 100% reliable as sites may store
+			// session tokens outside of user meta. However that does not lead
+			// to any problem, the user would simply be required to sign in
+			// again.
+			$current_user = null;
+			if ( (int) get_current_user_id() !== (int) $admin_user->ID ) {
+				$current_user = wp_get_current_user();
+			}
+			$user_sessions = get_user_meta( get_current_user_id(), 'session_tokens', true );
```

Comment:
`get_current_user_id()` always [return int value](https://developer.wordpress.org/reference/functions/get_current_user_id/). Remove duplicate call of function.
```suggestion
			$current_user    = null;
			$current_user_id = get_current_user_id();
			if ( $current_user_id !== (int) $admin_user->ID ) {
				$current_user = wp_get_current_user();
			}
			$user_sessions = get_user_meta( $current_user_id, 'session_tokens', true );
```